app/src/main/java/org/dhis2/usescases/searchTrackEntity/ui/mapper/TEICardMapper.kt

```diff
@@ -129,8 +129,13 @@ class TEICardMapper(
                 list = list,
                 enrolledPrograms = searchTEIModel.programInfo,
             )
+            val programUid: String = if (searchTEIModel.selectedEnrollment != null) {
+                searchTEIModel.selectedEnrollment.program().toString()
+            } else {
+                ""
```

Comment:
which are the consecuences of having programUid as an emtpy string?

Reply:
This is use to get a custom enrollmentLabel if there is one. If this is empty it will have the same impact as if it is null. It will show the default string "enrollment(s)"

Reply:
so, what do you think about setting it as null if we don't have a prograUid, I think considering empty string as null might cause unexpected flows that can be invisible for us.  What do you think?

Reply:
It's ok, I'll change so that the method checkEnrollmentStatus accepts null value as programUid